src/main/java/com/example/bookstore/model/Book.java

```diff
@@ -30,6 +35,11 @@ public class Book {
     private BigDecimal price;
     private String description;
     private String coverImage;
+    @ManyToMany(fetch = FetchType.EAGER)
+    @JoinTable(name = "books_categories",
+            joinColumns = @JoinColumn(name = "book_id"),
+            inverseJoinColumns = @JoinColumn(name = "category_id"))
+    private Set<Category> categories;
```

Comment:
Let's initialize it with new `HashSet` here as per our task description, just in case
```suggestion
    private Set<Category> categories = new HashSet();
```
Also, I would recommend excluding this field from equals hashcode & toString to avoid redundant db requests when we call those methods
The same usually goes for all related entities in Entity classes, depending on the situation